pkg/detectors/ftp/ftp.go

```diff
@@ -0,0 +1,103 @@
+package uri
+
+import (
+	"context"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/jlaffaye/ftp"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	allowKnownTestSites bool
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	keyPat = regexp.MustCompile(`\bftp:\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
```

Comment:
suggestion: You can remove a lot of those redundant escapes `\` as well as the `[]`

ex: `\bftp://\S{3,50}:(\S{3,50})@[-.%\w/:]+\` -> Gives you the same result.